src/state.rs

```diff
@@ -1,46 +1,120 @@
-use chrono::{DateTime, Local};
+use anyhow::Result;
+use tui::widgets::ListState;
 
 use crate::tasks::Task;
 
 use std::fs::{self, File, OpenOptions};
 
-#[derive(Default)]
 pub struct State {
```

Comment:
Looks like we have two classes responsible for stroing state of program runtime data (App, State). Do we need to? What are your reasoning for this seperation? Feel free to argue. If you agree then i propose to refactor this into one class (in seperate commit)

Reply:
sure, good idea

Reply:
my reasoning was to separate actions related to tasks-related data (such as adding, saving and loading them) from general app data - such as input mode